scene/2d/tile_map_layer.cpp

```diff
@@ -1677,21 +1686,15 @@ void TileMapLayer::_internal_update() {
 		dirty.flags[i] = false;
 	}
 
-	// List the cells to delete definitely.
-	Vector<Vector2i> to_delete;
+	// Remove cells that are empty after the cleanup.
 	for (SelfList<CellData> *cell_data_list_element = dirty.cell_list.first(); cell_data_list_element; cell_data_list_element = cell_data_list_element->next()) {
 		CellData &cell_data = *cell_data_list_element->self();
 		// Select the cell from tile_map if it is invalid.
 		if (cell_data.cell.source_id == TileSet::INVALID_SOURCE) {
-			to_delete.push_back(cell_data.coords);
+			tile_map.erase(cell_data.coords);
 		}
 	}
 
-	// Remove cells that are empty after the cleanup.
-	for (const Vector2i &coords : to_delete) {
-		tile_map.erase(coords);
-	}
-
```

Comment:
That's kinda unrelated change, is there any reason why it was done like that? :thinking: (could revert that part if so)